spec/units-of-measure.md

```diff
@@ -1,138 +1,133 @@
+    int32 < measure-literal >                   -- 32 - bit integer constant
+    int64 < measure-literal >                   -- 64 - bit integer constant
+    ieee32 < measure-literal >                  -- single-precision float32 constant
+    ieee64 < measure-literal >                  -- double-precision float constant
+    decimal < measure-literal >                 -- decimal constant
 
 measure-atom :=
-    typar -- variable measure, such as 'U
-    long-ident -- named measure, such as kg
-    ( measure-simp )            -- parenthesized measure, such as (N m)
+    typar                                       -- variable measure, such as 'U
+    long-ident                                  -- named measure, such as kg
+    ( measure-simp )                            -- parenthesized measure, such as (N m)
 
 measure-power :=
     measure-atom
-    measure-atom ^ int32        -- power of measure, such as m^3
+    measure-atom ^ int32                        -- power of measure, such as m^3
 
 measure-seq :=
     measure-power
     measure-power measure-seq
 
 measure-simp :=
-    measure-seq                 -- implicit product, such as 'U 'V^3
-    measure-simp * measure-simp -- product, such as 'U * 'V
-    measure-simp / measure-simp -- quotient, such as 'U / 'V
-    / measure-simp              -- reciprocal, such as /'U
-    1                           -- dimensionless measure (no units)
+    measure-seq                                 -- implicit product, such as 'U 'V^3
+    measure-simp * measure-simp                 -- product, such as 'U * 'V
+    measure-simp / measure-simp                 -- quotient, such as 'U / 'V
+    / measure-simp                              -- reciprocal, such as /'U
+    1                                           -- dimensionless measure (no units)
 
 measure :=
-    _                           -- anonymous measure
-measure-simp                    -- simple measure, such as 'U 'V
+    _                                           -- anonymous measure
+measure-simp                                    -- simple measure, such as 'U 'V
 ```
-Measure definitions use the special `Measure` attribute on type definitions. Measure parameters use
-the syntax of generic parameters with the same special `Measure` attribute to parameterize types and
-members by units of measure. The primitive types `sbyte`, `int16`, `int32`, `int64`, `float`, `float32`, and
-`decimal` have non-parameterized (dimensionless) and parameterized versions.
+
+Measure definitions use the special `Measure` attribute on type definitions. Measure parameters use the syntax of generic parameters with the same special `Measure` attribute to parameterize types and members by units of measure. The primitive types `byte`, `sbyte`, `int16`, `int32`, `int64`, `float`, `float32`, `decimal`, `unativeint`, and `nativeint` have non-parameterized (dimensionless) and parameterized versions.
```

Comment:
It seems the phrase "with the same special `Measure` attribute" in the second sentence should be deleted.
And in the list of types the uint* types are missing.

Reply:
I tried making the second occurrance parenthetical, since it's important to capture that the required form is, eg:

```fsharp
[<Measure>]
type MyType<[<Measure>] 'U> = ...
```

When I tried simply removing the phrase, it read as though the following would be valid:

```fsharp
[<Measure>]
type MyType<'U> = ...
```

which is not a compiler error, but will also not do what the developer expects with regards to UoM.

Of course, I am also more-than-happy to try re-wording things further (technical writing is hard 🙂).

Reply:
Ah, I only now understand this.
Maybe add the above example to the text?

Reply:
Actually it should then be an example only for the second occurrence, like `type MyType<[<Measure>] 'U> = int<'U>
`